RingBufCPP.h

```diff
@@ -70,14 +114,47 @@ bool pull(Type *dest)
     return ret;
 }
 
+/**
+* Remove "length" last elements from buffer, and copy them to "dest" buffer
+* Return: number of elements actually copied (could be less if numElements < length).
+*/
+size_t pull(Type *dest, size_t length)
```

Comment:
Is there a benefit to this over just calling `pull()` N times? With this approach you have an O(N) critical section which is not super ideal.

Reply:
How could the performance of making N pulls() (and calculating the tail each time with % operator!) be better than just making a memcpy to the dest array? I don't get it. The concern about memcpy not calling the copy constructor is a valid point.

Reply:
Left some comments about this below, but to answer you questions:
- % and / is basically free if MaxElements is a power of 2 (which should be the case).
- Performance is not the only consideration on embedded systems. Latency/responsiveness is generally more important, and this change masks interrupts for O(n), increasing interrupt response latency.